crates/astria-sequencer/src/fees/query.rs

```diff
@@ -240,184 +257,155 @@ pub(crate) async fn get_fees_for_transaction<S: StateRead>(
+                    get_or_init_fees::<BridgeLock, _, _>(state, &bridge_lock_fees, "BridgeLock")
+                        .await?;
                 calculate_and_add_fees(
                     act,
                     act.fee_asset.to_ibc_prefixed(),
                     &mut fees_by_asset,
-                    bridge_lock_fees.base,
-                    bridge_lock_fees.multiplier,
+                    bridge_lock_fees,
                 );
             }
             Action::BridgeUnlock(act) => {
-                let bridge_unlock_fees = bridge_unlock_fees
-                    .get_or_try_init(|| async { state.get_bridge_unlock_fees().await })
-                    .await
-                    .wrap_err("failed to get bridge unlock fees")?
-                    .ok_or_eyre("fees not found for `BridgeUnlock` action, hence it is disabled")?;
+                let bridge_unlock_fees = get_or_init_fees::<BridgeUnlock, _, _>(
+                    state,
+                    &bridge_unlock_fees,
+                    "BridgeUnlock",
+                )
+                .await?;
                 calculate_and_add_fees(
                     act,
                     act.fee_asset.to_ibc_prefixed(),
                     &mut fees_by_asset,
-                    bridge_unlock_fees.base,
-                    bridge_unlock_fees.multiplier,
+                    bridge_unlock_fees,
                 );
             }
             Action::BridgeSudoChange(act) => {
-                let bridge_sudo_change_fees = bridge_sudo_change_fees
-                    .get_or_try_init(|| async { state.get_bridge_sudo_change_fees().await })
-                    .await
-                    .wrap_err("failed to get bridge sudo change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `BridgeSudoChange` action, hence it is disabled",
-                    )?;
+                let bridge_sudo_change_fees = get_or_init_fees::<BridgeSudoChange, _, _>(
+                    state,
+                    &bridge_sudo_change_fees,
+                    "BridgeSudoChange",
+                )
+                .await?;
                 calculate_and_add_fees(
                     act,
                     act.fee_asset.to_ibc_prefixed(),
                     &mut fees_by_asset,
-                    bridge_sudo_change_fees.base,
-                    bridge_sudo_change_fees.multiplier,
+                    bridge_sudo_change_fees,
                 );
             }
             Action::ValidatorUpdate(_) => {
-                validator_update_fees
-                    .get_or_try_init(|| async { state.get_validator_update_fees().await })
-                    .await
-                    .wrap_err("failed to get validator update fees")?
-                    .ok_or_eyre(
-                        "fees not found for `ValidatorUpdate` action, hence it is disabled",
-                    )?;
+                get_or_init_fees::<ValidatorUpdate, _, _>(
```

Comment:
it doesn't look like there's a point to calling `get_or_init_fees` in this action or any of the ones below that are feeless, as they don't update `fees_by_asset`, which is the point of this function

Reply:
This was done so that the check would fail if the transaction would fail during execution (due to a given action not having fees). I think it may be worth it to keep this logic since it is used in mempool as well as `check_and_execute` and may catch a failure-bound transaction before it gets executed.